Findings Overview Page HTML

alexsoble opened the pull request:
# Notes 

+ Extracts out the "logged in user header" to an HTML template that can be reused. 
+ Aiming for an HTML-only mockup of the Findings Overview Page; let's save dynamic data population for a separate PR. 
+ Making an attempt at readability but not aesthetic perfection on mobile. There aren't designs for mobile states, so I'm winging it.

# Screenshot

<img width="1277" alt="findings-overview-desktop-screenshot" src="https://user-images.githubusercontent.com/3209501/60285588-dbc97080-98c2-11e9-88e7-72102af79715.png">

alexsoble commented:
Progress so far! 

<img width="1274" alt="findings-overview-progress-so-far" src="https://user-images.githubusercontent.com/3209501/60134685-aeb37b80-9765-11e9-8dc8-4055e6ad10e0.png">

alexsoble commented:
Interactions shown in the Sketch design:

1. Tooltip hover over the oval user label showing more information.
2. Toggle sort direction.
3. Customize sort.

alexsoble commented:
🤔 🤔 I wonder if this PR is ready for review now. We could ship this as a first pass containing static HTML only, with follow-on PRs adding interactions and data pulled from the database.

fureigh commented on single-audit/resolve_findings/templates/resolve_findings/finding-resolution-page.html:
```diff
@@ -2,20 +2,7 @@
-      <span class="text-middle">Tom Hammersmith</span>
-      <img alt="user icon"
-           src="{% static 'img/user-circle-gray.svg' %}"
-           height="24"
```
Is specifying a static number (`24`) kosher here?

alexsoble replied:
@fureigh Hm, I don't see why not. It's a small enough icon, I don't think it needs to be sized differently at different screen widths. Open to suggestions for how to size this more elegantly or idiomatically though — I want to learn!

fureigh commented:
@alexsoble Sounds reasonable! Data's being pulled from the database currently, but it'll need to be made more granular. Assign review to me whenever you're ready.

alexsoble commented:
@fureigh Sounds good! I'll add one more row to the table with a couple of different states/permutations included, and then tag you for review.

alexsoble commented:
Screenshot of current state on desktop: 

<img width="1277" alt="findings-overview-desktop-screenshot" src="https://user-images.githubusercontent.com/3209501/60285735-061b2e00-98c3-11e9-97bc-09da8f98a234.png">

alexsoble commented:
☝️ I think the current state of the HTML is fairly faithful to the Sketch designs for the desktop view. There aren't any designs for mobile views, and at the moment the HTML isn't highly mobile responsive. Here's what the page would look like on a Galaxy S5 right now:

<img width="340" alt="findings-overview-galaxy-s5-table" src="https://user-images.githubusercontent.com/3209501/60286855-53000400-98c5-11e9-8fcb-d70f896ec912.png">
<img width="319" alt="findings-overview-galaxy-s5-filtering" src="https://user-images.githubusercontent.com/3209501/60286856-53989a80-98c5-11e9-94a5-d6f3283a07e2.png">

@fureigh and @bpdesigns, what are your thoughts and feelings on mobile responsiveness? I can rewrite the table to use flexbox so that information renders horizontally on desktop and tablets and then switches to vertically on mobile widths. Might be the last front-end improvement that fits into this sprint.

bpdesigns commented:
Sounds good to me

alexsoble commented:
So, I ultimately didn't end up changing the structure of the table at mobile widths — I cleaned up a bunch of margins and paddings that looked nice on desktop but were creating issues on mobile; also added a few handy `@media` queries. 

Here's a GIF of the mobile view now:

![galaxy-s5-more-responsive](https://user-images.githubusercontent.com/3209501/60292156-5c8f6900-98d1-11e9-8a2a-19fb9dcefdbe.gif)

alexsoble commented:
... plus, here's the latest on desktop. I changed around the tab navigation styles a bit so that they stay understandable on mobile (the white background of the selection button was bleeding into the white background of the app on mobile, making the structure hard to understand):

<img width="1273" alt="latest-desktop" src="https://user-images.githubusercontent.com/3209501/60292296-a1b39b00-98d1-11e9-8a02-09e5590ecc33.png">

alexsoble commented:
@fureigh, I think this should be in a good place for review! @bpdesigns, consider yourself an optional reviewer 😄

fureigh commented:
Thanks so much for all of this, @alexsoble! This is a fantastic foundation to work from.

The display's a little broad in Chrome, but that's simple enough to change — I'm mentioning it here as a note for the future.

I reincorporated dynamic display of actual findings, so as to not have that functionality reversed by this PR.

:shipit:❗️
